### PR TITLE
Woogles Errors

### PR DESCRIPTION
**Pull request by jvc56**

### CODE REVIEW

**Review by domino14 — APPROVED** [review]
this seems like a good start.

**Review by atomiccupcake — APPROVED** [review]

File: pkg/tournament/tournament.go [hunk 1]
```diff
@@ -140,14 +140,14 @@ func SetTournamentMetadata(ctx context.Context, ts TournamentStore, meta *pb.Tou
...
-		return fmt.Errorf("tournament %s has finished", meta.Id)
+		return entity.NewWooglesError(realtime.WooglesError_TOURNAMENT_FINISHED, []string{t.Name, ""})
 	}
 
 	t.Lock()
 	defer t.Unlock()
 	name := strings.TrimSpace(meta.Name)
 	if name == "" {
-		return errors.New("name cannot be blank")
+		return entity.NewWooglesError(realtime.WooglesError_TOURNAMENT_EMPTY_NAME, []string{t.Name, ""})
```
Comment (domino14) [inline pkg/tournament/tournament.go:150]: how come this and line 143 have an extra empty string?

Comment (jvc56) [reply pkg/tournament/tournament.go:150]: good question

**Comment by domino14** [comment]
could u rebase it?

**Comment by domino14** [comment]
@jvc56 i ran into a bug while testing this. Apparently, you can submit a result for a tournament as follows:

- p1 is playing p2
- submit result: p1 beat p1 350-300 (does not get rejected)
- shows up as p1 beating p2 350-300

**Comment by domino14** [comment]
Another bug:

- Four players in a division, Fontes pairings
- Remove one player after starting round 1
- PairRound fails so the player is not removed in the front end
- Upon refreshing, re-add the player we removed
- Error when adding player to "registrants" table (probably because we didn't remove player from registrants since step 3 failed, and it wasn't a transaction)

**Comment by domino14** [comment]
close #888

**Pull request merged**